statsmodels/stats/proportion.py

```diff
@@ -1044,3 +1061,802 @@ def proportions_chisquare_pairscontrol(count, nobs, value=None,
+                statistic, pvalue = res[:2]
+            distr = 'normal'
+            diff_stat = None
+
+        else:
+            raise ValueError('method not recognized')
+
+    elif compare == 'odds-ratio':
+
+        if method in ['logit', 'logit-adjusted', 'logit-smoothed']:
+            if method in ['logit-smoothed']:
+                adjusted = _shrink_prob(count1, nobs1, count2, nobs2,
+                                       shrink_factor=2, return_corr=False)[0]
+                count1_, nobs1_, count2_, nobs2_ = adjusted
+
+            else:
+                addhalf = 0.5 if method == 'logit-adjusted' else 0
+                count1_, nobs1_ = count1 + addhalf, nobs1 + 2 * addhalf
+                count2_, nobs2_ = count2 + addhalf, nobs2 + 2 * addhalf
+            p1_ = count1_ / nobs1_
+            p2_ = count2_ / nobs2_
+            odds_ratio_ = p1_ / (1 - p1_) / p2_ * (1 - p2_)
+            var = (1 / count1_ + 1 / (nobs1_ - count1_) +
+                   1 / count2_ + 1 / (nobs2_ - count2_))
+
+            diff_stat = np.log(odds_ratio_) - np.log(value)
+            statistic = diff_stat / np.sqrt(var)
+            distr = 'normal'
+
+        elif method == 'score':
+            res = score_test_proportion_2indep(count1, nobs1, count2, nobs2,
+                                              value=value, compare=compare,
+                                              alternative=alternative,
+                                              correction=correction,
+                                              return_results=return_results)
+            if return_results is False:
+                statistic, pvalue = res[:2]
+            distr = 'normal'
+            diff_stat = None
+        else:
+            raise ValueError('method "%s" not recognized' % method)
+
+    else:
+        raise ValueError('compare "%s" not recognized' % compare)
+
+    if distr == 'normal' and diff_stat is not None:
+        statistic, pvalue = _zstat_generic2(diff_stat, np.sqrt(var),
+                                            alternative=alternative)
+
+    if return_results:
+        if res is None:
+            res = Holder(statistic = statistic,
+                         pvalue = pvalue,
+                         compare = compare,
+                         method = method,
+                         diff = diff,
+                         ratio = ratio,
+                         odds_ratio = odds_ratio,
+                         variance = var,
+                         alternative = alternative,
+                         )
+        else:
+            # we already have a return result from score test
+            # add missing attributes
+            res.diff = diff
+            res.ratio = ratio
+            res.odds_ratio = odds_ratio
+        return res
+    else:
+        return statistic, pvalue
+
+
+def _std_2prop_power(diff, p2, ratio=1, alpha=0.05, value=0):
+    """compute standard error under null and alternative for 2 proportions
+
```

Comment:
only for "diff" ?
nobs_ratio would be more explicit, if it is ratio of nobs

Reply:
in power computation, I always use ratio for nobs-ratio.

but for 2sample proportion, we also have risk-ratio and odds-ratio